apps/framework-docs/src/pages/building/make-changes.mdx

```diff
@@ -0,0 +1,20 @@
+# Changing your data models
+Change management in your data stack is critical to ensure that your data is always accurate and up to date based on your end user's expectations. Broken dashboards, reports, or machine learning models can to unhappy customers and lost revenue. MooseJS makes it easy to change your data models and ensure that your data stack is always up to date. To do so, we leverage git workflows to ensure that changes are tracked and can be rolled back if necessary.
+
+## Adding a new data model
+Adding a new data source is as simple as adding a new model to an existing or new file in the `data models` directory. MooseJS will automatically detect the new model and add it to the data stack. 
+
+## Changing an existing data model
+
+### During development
+Whenever you save a file, Moose will automatically detect the change and update your data stack. Under the hood, MooseJS manages the state of your stack and creates temporary tables each time you save your file. This allows you to test your changes without worrying the effects on the existing data. 
+
+Once you commit your changes, MooseJS will automatically create a new version of your data stack and turn on the migration mechanisms that ensure that historical data is present in your modified schema. The migration mechanisms assume that the initial state of your data starts at the initial state of your branch. This limits the number of active migrations that are running in your stack to optimize performance and costs.
```

Comment:
Not sure what you mean by that. Why isn't the migration base state the previous commit?

The advantage of using a commit is that it is the same simple predictable process on the `main` branch or on feature branches

Reply:
Because you'd need a new infra + migration mechanism for each step no?

Reply:
Yeah - as long as we have a way to prevent it from getting too large on disk that should be ok no?